src/refdb_fs.c

```diff
@@ -362,6 +364,13 @@ static const char *loose_parse_symbolic(git_buf *file_content)
 	return refname_start;
 }
 
+static bool is_per_worktree_ref(const char *ref_name)
+{
+	return strcmp("HEAD", ref_name) == 0 ||
+	    strcmp("MERGE_HEAD", ref_name) == 0 ||
+	    strcmp("ORIG_HEAD", ref_name) == 0;
```

Comment:
Rather than hardcoding the list of refs that belong in the worktree rather than the common dir, could you generalize this to allow access to other refs in the worktree?  This hardcoded list misses FETCH_HEAD, which comes from the worktree.  And other applications may want to have their own special worktree-state refs similar to HEAD.

I'd suggest treating any top-level ref (not starting with `refs/`) as going in the worktree, at least for any function that doesn't today try to DWIM and prepend `refs/heads` or similar itself.